builtins/web/fetch/request-response.cpp

```diff
@@ -497,9 +495,11 @@ bool RequestOrResponse::append_body(JSContext *cx, JS::HandleObject self, JS::Ha
 JSObject *RequestOrResponse::headers(JSContext *cx, JS::HandleObject obj) {
   JSObject *headers = maybe_headers(obj);
   if (!headers) {
-    host_api::HttpHeadersGuard guard = Request::is_instance(obj)
-      ? host_api::HttpHeadersGuard::Request
-      : host_api::HttpHeadersGuard::Response;
+    // Incoming request and incoming response headers are immutable per service worker
```

Comment:
I understand this is making the implementation spec compliant which is good, but this is also a breaking change as some applications will now throw an error because of this change. With that in my, I think this should be released as a new major version (if not in StarlingMonkey then in downstream projects)

Reply:
Since this is implemented in the `request-response` usage of Headers, which Fastly maintains its own version of, we can choose to not implement this yet in js-compute-runtime to maintain this as a non-breaking change.

I'll make sure to include this in the Fastly PR at https://github.com/fastly/js-compute-runtime/pull/844.